rust/stackable-cockpit/src/utils/k8s/client.rs

```diff
@@ -383,6 +385,43 @@ impl Client {
         endpoints_api.get(name).await.context(KubeClientFetchSnafu)
     }
 
+    /// Try to resolve the given [`GroupVersionKind`]. In case the resolution fails a discovery is run to pull in new
+    /// GVKs that are not present in the [`Discovery`] cache. Afterwards a normal resolution is issued.
+    async fn resolve_gvk(
+        &self,
+        gvk: &GroupVersionKind,
+    ) -> Result<Option<(ApiResource, ApiCapabilities)>> {
+        let resolved = self.discovery.read().unwrap().resolve_gvk(gvk);
+
+        Ok(match resolved {
+            Some(resolved) => Some(resolved),
+            None => {
+                tracing::warn!(?gvk, "Discovery did not include gvk, re-running discovery");
```

Comment:
nit: Tracing events should say what happened and not what will happen. I suggest:

```suggestion
                tracing::warn!(?gvk, "discovery did not include gvk");
```

Then instrumenting the discovery function which would then have an event on whether it ran or didn't run.

---

This can be difficult at times though, eg: moving an implcit return into a variable so that you can choose the event level before finally returning (see the `run_discovery` function).

Reply:
Changed to `info!(?gvk, "discovery did not include gvk");` in fb9ea43

Reply:
I commented somewhere, but I don't see it, so will write it here:

nit (non-blocker): I would prefer we qualify tracing stuff with `tracing::`. It just helps it stick out. There is nothing in the style-guide for this, so feel free to disregard for now (just voicing an opinion).

Reply:
Funny thing I had the same discussion with @Techassi yesterday and we prefer `info!()` :D
I guess it makes sense to start a decision on this and add it to our style guide. But I wouldn't block this PR on that

Reply:
Decision: https://github.com/stackabletech/decisions/issues/15